### PR TITLE
Persistent Tabs and some bug fixes

### PR DESCRIPTION
**Pull request by mahmoud-adam85**

### CODE REVIEW

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 1]
```diff
@@ -3812,6 +3818,16 @@ extension BrowserViewController: KeyboardHelperDelegate {
     }
 }
 
+extension BrowserViewController: SessionRestoreHelperDelegate {
```
Comment (naira-cliqz) [inline Client/Frontend/Browser/BrowserViewController.swift:3829]: Let's move this extension to CliqzBrowserViewController and remove  updateUIForReaderHomeStateForTab part as we discussed

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 2]
```diff
@@ -573,22 +571,17 @@ class BrowserViewController: UIViewController {
...
 /*
         if PLCrashReporter.sharedReporter().hasPendingCrashReport() {
             PLCrashReporter.sharedReporter().purgePendingCrashReport()
             showRestoreTabsAlert()
         } else {
+*/
             log.debug("Restoring tabs.")
             tabManager.restoreTabs()
             log.debug("Done restoring tabs.")
-        }
```
Comment (naira-cliqz) [inline Client/Frontend/Browser/BrowserViewController.swift:585]: Let's remove Cliqz related commented code.

File: Client/Cliqz/Frontend/Browser/Dashboard/TabsViewController.swift [hunk 3]
```diff
@@ -151,7 +151,7 @@ extension TabsViewController: UITableViewDataSource, UITableViewDelegate {
         let freshtab = tab.displayURL?.absoluteString?.contains("cliqz/goto.html") ?? false
         
 		if tab.displayURL != nil && !freshtab{
-            cell.titleLabel.text = tab.title != nil && tab.title != "" ? tab.title : tab.displayURL?.absoluteString
+            cell.titleLabel.text = tab.displayTitle != "" ? tab.displayTitle : tab.displayURL?.absoluteString
```
Comment (naira-cliqz) [inline Client/Cliqz/Frontend/Browser/Dashboard/TabsViewController.swift]: text should be assigned just tab.displayTitle. DisplayTitle has already the check for absoluteString.

**Review by naira-cliqz — CHANGES_REQUESTED** [review]

File: Client/Frontend/Browser/BrowserViewController.swift (hunk 1 above)
Comment (mahmoud-adam85) [reply Client/Frontend/Browser/BrowserViewController.swift:3829]: this is FireFox code as I cherry-pick this commit

File: Client/Frontend/Browser/BrowserViewController.swift (hunk 2 above)
Comment (mahmoud-adam85) [reply Client/Frontend/Browser/BrowserViewController.swift:585]: it is already removed

**Comment by mahmoud-adam85** [comment]
@naira-cliqz done with all changes

**Pull request merged**